dali/python/nvidia/dali/_debug_mode.py

```diff
@@ -395,11 +394,46 @@ def _check_batch_size(self, classification, input_idx):
             self._pipe._cur_iter_batch_info.check_input(
                 len(classification.data), self._source_context, self._op_name, input_idx)
 
+    def _check_call_arg_meta_data(self, expected_data, actual_data, arg_type, value):
+        """ Check for changes in layout, ndim and dtype. 
+        
+        Args:
+            expected_data: Expected value of the data.
+            actual_data: Actual value of the data.
+            arg_type (str): String representation of the argument type, e.g. 'Input', 'Argument'.
+            value (str): Argument name for keyword arguments and a number for inputs.
+        """
+
+        def raise_err(meta_name, actual_value, expected_value):
+            raise RuntimeError(
+                f"{arg_type} {value} for operator '{self._op_name}' has "
+                f"{meta_name} = {actual_value}, expected: {expected_value}.")
+
+        expected_input_set = isinstance(expected_data, list)
+        if expected_input_set != isinstance(actual_data, list):
+            raise RuntimeError(f"{arg_type} {value} expected {'' if expected_input_set else 'not '}"
+                               f"to be an input set.")
+
+        if isinstance(actual_data, list):
+            # Checking input set.
+            for expected_elem, actual_elem in zip(expected_data, actual_data):
```

Comment:
Is it possible for the two list to have different lengths making this check doing only part of its job?

Reply:
Added check for that.